### PR TITLE
Move mesh

### PR DESCRIPTION
**Pull request by liamhuber**
Heads up @raynol-dsouza that in the process of moving `RectMesh` I discovered that the last PR introduced a bug into the laplacian -- it worked OK for perfectly smooth functions, but has numeric trouble for noisy fields, which is exactly our starting point in `Schroedinger`.

EDIT:
Sorry, I intended for this to be a pure refactor:move, but I found a bug in `laplacian` and got carried away. In addition to moving the laplacian bug is squashed, accuracy (i.e. points in the stencil) is handled as a class attribute, and there's a demo notebook.

### CODE REVIEW

File: pyiron_continuum/mesh.py [hunk 1]
```diff
@@ -107,6 +119,7 @@ def __init__(
             self,
             bounds: Union[float, int, BoundsList, np.ndarray] = 1,
             divisions: Union[int, List[int], np.ndarray] = 1,
+            accuracy: int = 2,
```
Comment (samwaseda) [inline pyiron_continuum/mesh.py:122]: It's written default is 4 below

Comment (liamhuber) [reply pyiron_continuum/mesh.py:122]: good eye. Will let the tests finish then push the patch.

**Comment by samwaseda** [comment]
Currently I see that `derivative` adds a new axis in the first column, so from `(nx, ny, nz)` it makes `(3, nx, ny, nz)`, correct? It's super important to document it somewhere in the DocString because other people (e.g. me) might do it otherwise other way around (i.e. `(nx, ny, nz, 3)`).

**Comment by liamhuber** [comment]
> @liamhuber
> 
> Aha. Good catch! I cannot test the branch unless it has #63 merged (because of the fenics `ModuleImportError`. Could you please merge it into this PR?

Once #63 is merged into master I'll pull master in. I don't want to mess around with branch-to-branch merges atm. There is no rush to get this stuff in though, so I don't mind waiting for 63.

**Comment by samwaseda** [comment]
I just realised that `RectMesh` is in `schroedinger`, but doesn't it make more sense to have it in its own thing (especially since now I'm thinking about using it for my grain boundary dynamics stuff)

**Comment by samwaseda** [comment]
> > I just realised that `RectMesh` is in `schroedinger`, but doesn't it make more sense to have it in its own thing (especially since now I'm thinking about using it for my grain boundary dynamics stuff)
> 
> Take a quick peek at the PR name 😉 (I have to confess, when I looked at the GitHub diff it did not pop out to me even when searching for it!)

XD I thought this branch had already been merged. Typical mistake of mine sorry!!!

**Pull request merged**